include/mbgl/storage/default_file_source.hpp

```diff
@@ -18,8 +20,13 @@ class DefaultFileSource : public FileSource {
 
     std::unique_ptr<FileRequest> request(const Resource&, Callback) override;
 
+    // For testing purposes only.
```

Comment:
We could either to forward declarations in the global namespace, and use `friend class ::Storage_Foo_Test` or something like this:

```cpp
#ifdef GTEST_TEST
public:
#else
private:
#endif
```